### PR TITLE
Add table aws_macie2_classification_job. closes #459

### LINKED ISSUE #459: Add table aws_macie2_classification_job
**References**
`Terraform`
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/macie2_classification_job

`AWS SDK`
https://docs.aws.amazon.com/macie/latest/APIReference/jobs-list.html#jobs-listpost
https://docs.aws.amazon.com/macie/latest/APIReference/jobs-jobid.html#jobs-jobidget

**Requirements**
Need this table for one of the CIS requirements
`Ensure all data in Amazon S3 has been discovered, classified and secured when required.`

Perform the following steps to determine if Macie is running:

From Console:
```
1. Login to the Macie console at https://console.aws.amazon.com/macie/ 
2. In the left hand pane click on By job under findings.
3. Confirm that you have a Job setup for your S3 Buckets
```

When you log into the Macie console if you aren't taken to the summary page and you don't have a job setup and running then refer to the remediation procedure below.
If you are using a 3rd Party tool to manage and protect your s3 data you meet this recommendation.

### PR DESCRIPTION
**Pull request by Priyanka585464**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_macie2_classification_job []

PRETEST: tests/aws_macie2_classification_job

TEST: tests/aws_macie2_classification_job
Running terraform
aws_macie2_account.test: Creating...
aws_s3_bucket.test: Creating...
aws_macie2_account.test: Creation complete after 2s [id=013122550996]
aws_s3_bucket.test: Still creating... [10s elapsed]
aws_s3_bucket.test: Still creating... [20s elapsed]
aws_s3_bucket.test: Creation complete after 23s [id=turbottest94345]
aws_macie2_classification_job.named_test_resource: Creating...
aws_macie2_classification_job.named_test_resource: Creation complete after 3s [id=41eb98181348afb949d8c8f9d33bd142]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

account_id = "013122550996"
aws_partition = "aws"
aws_region = "us-east-1"
resource_aka = "arn:aws:macie2:us-east-1:013122550996:classification-job/41eb98181348afb949d8c8f9d33bd142"
resource_id = "41eb98181348afb949d8c8f9d33bd142"
resource_name = "turbottest94345"

Running SQL query: test-get-query.sql
[
  {
    "arn": "arn:aws:macie2:us-east-1:013122550996:classification-job/41eb98181348afb949d8c8f9d33bd142",
    "name": "turbottest94345",
    "partition": "aws",
    "region": "us-east-1",
    "tags": {
      "Name": "turbottest94345"
    }
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:macie2:us-east-1:013122550996:classification-job/41eb98181348afb949d8c8f9d33bd142",
    "job_id": "41eb98181348afb949d8c8f9d33bd142",
    "name": "turbottest94345",
    "partition": "aws",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "013122550996",
    "region": "us-east-1",
    "title": "turbottest94345"
  }
]
✔ PASSED

POSTTEST: tests/aws_macie2_classification_job

TEARDOWN: tests/aws_macie2_classification_job

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  job_id,
  arn,
  name,
  job_status,
  region
from
  aws_macie2_classification_job;
```
```
+----------------------------------+-------------------------------------------------------------------------------------------+--------+------------+-----------+
| job_id                           | arn                                                                                       | name   | job_status | region    |
+----------------------------------+-------------------------------------------------------------------------------------------+--------+------------+-----------+
| 65af18dd4760059004c2a2e2aad203a2 | arn:aws:macie2:us-east-1:986325076436:classification-job/65af18dd4760059004c2a2e2aad203a2 | steam1 | RUNNING    | us-east-1 |
+----------------------------------+-------------------------------------------------------------------------------------------+--------+------------+-----------+
```

### List jobs which are Paused/Cancelled

```sql
select
  job_id,
  arn,
  name,
  job_status as status
from
  aws_macie2_classification_job
where
  job_status = 'CANCELLED'
  or job_status = 'PAUSED';
```
```
+--------+-----+------+--------+
| job_id | arn | name | status |
+--------+-----+------+--------+
+--------+-----+------+--------+
```

### Get S3 bucket details where job is running

```sql
select
  job_id,
  detail -> 'AccountId' as account_id,
  detail -> 'Buckets' as buckets
from
  aws_macie2_classification_job,
  jsonb_array_elements(s3_job_definition -> 'BucketDefinitions') as s3_details,
  jsonb(s3_details) as detail;
```
```
+----------------------------------+----------------+------------------------------------------------------------+
| job_id                           | account_id     | buckets                                                    |
+----------------------------------+----------------+------------------------------------------------------------+
| 65af18dd4760059004c2a2e2aad203a2 | "986325076436" | ["appstream-app-settings-us-east-1-986325076436-vz8cc0a5"] |
+----------------------------------+----------------+------------------------------------------------------------+
```

### Check number of times that job has run

```sql
select
  job_id,
  arn,
  statistics ->> 'ApproximateNumberOfObjectsToProcess' as approximate_number_of_objects_to_process,
  statistics ->> 'NumberOfRuns' as number_of_runs
from
  aws_macie2_classification_job;
```
```
+----------------------------------+-------------------------------------------------------------------------------------------+------------------------------------------+----------------+
| job_id                           | arn                                                                                       | approximate_number_of_objects_to_process | number_of_runs |
+----------------------------------+-------------------------------------------------------------------------------------------+------------------------------------------+----------------+
| 65af18dd4760059004c2a2e2aad203a2 | arn:aws:macie2:us-east-1:986325076436:classification-job/65af18dd4760059004c2a2e2aad203a2 | 0                                        | 0              |
+----------------------------------+-------------------------------------------------------------------------------------------+------------------------------------------+----------------+
```

### CODE REVIEW

File: aws-test/tests/aws_macie2_classification_job/variables.tf [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+resource "aws_macie2_classification_job" "named_test_resource" {
+  depends_on = [aws_macie2_account.test]
+  job_type = "ONE_TIME"
+  name     = var.resource_name
+  s3_job_definition {
+    bucket_definitions {
+      account_id = "986325076436"
+      buckets = ["appstream-app-settings-us-east-1-986325076436-vz8cc0a5"]
```
Comment (Subhajit97) [inline aws-test/tests/aws_macie2_classification_job/variables.tf:66]: Can we remove the reference of existing bucket, and create a new bucket for test?

File: aws/table_aws_macie2_classification_job.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+)
+
+//// TABLE DEFINITION
+
+func tableAwsMacie2ClassificationJob(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_macie2_classification_job",
+		Description: "AWS Macie2 Classification Job",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.AllColumns([]string{"Job_id"}),
```
Comment (Subhajit97) [inline aws/table_aws_macie2_classification_job.go:20]:
```suggestion
			KeyColumns:        plugin.AllColumns([]string{"job_id"}),
```

File: aws/table_aws_macie2_classification_job.go [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+			ShouldIgnoreError: isNotFoundError([]string{"AccessDeniedException"}),
```
Comment (Subhajit97) [inline aws/table_aws_macie2_classification_job.go:21]: Generally we don't hide `AccessDeniedException` error

File: aws/table_aws_macie2_classification_job.go [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+			},
+			{
+				Name:        "tags_src",
+				Description: "The list of tags for the classification job.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsMacie2ClassificationJob,
+				Transform:   transform.FromField("Tags"),
+			},
+
+			// Standard columns for all tables
```
Comment (Subhajit97) [inline aws/table_aws_macie2_classification_job.go:127]:
```suggestion
			// Steampipe standard columns
```

File: aws/table_aws_macie2_classification_job.go [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsMacie2ClassificationJob,
+				Transform:   transform.FromField("Tags"),
```
Comment (Subhajit97) [inline aws/table_aws_macie2_classification_job.go:139]:
```suggestion
```

Comment (Subhajit97) [reply aws/table_aws_macie2_classification_job.go:139]: What is datatype for `tags` in response?
Is it `key-value` object? 
If yes, can we remove `tags_src` column?

File: aws/table_aws_macie2_classification_job.go [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsMacie2ClassificationJob,
+				Transform:   transform.FromField("JobArn").Transform(arnToAkas),
```
Comment (Subhajit97) [inline aws/table_aws_macie2_classification_job.go:146]:
```suggestion
				Transform:   transform.FromField("JobArn").Transform(transform.EnsureStringArray),
```

File: aws/table_aws_macie2_classification_job.go [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+//// LIST FUNCTION
+
+func listAwsMacie2ClassificationJobs(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	plugin.Logger(ctx).Trace("listAwsMacie2ClassificationJob", "AWS_REGION", region)
```
Comment (Subhajit97) [inline aws/table_aws_macie2_classification_job.go:161]:
```suggestion
	plugin.Logger(ctx).Trace("listAwsMacie2ClassificationJobs", "AWS_REGION", region)
```

File: aws/table_aws_macie2_classification_job.go [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+
+	// id := d.KeyColumnQuals["job_id"].GetStringValue()
+
```
Comment (Subhajit97) [inline aws/table_aws_macie2_classification_job.go:164]:
```suggestion
```

File: aws/table_aws_macie2_classification_job.go [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+		return nil, err
+	}
+
+	// List call
+	err = svc.ListClassificationJobsPages(
+		&macie2.ListClassificationJobsInput{},
+		func(page *macie2.ListClassificationJobsOutput, isLast bool) bool {
+			for _, job := range page.Items {
+				d.StreamListItem(ctx, job)
+
```
Comment (Subhajit97) [inline aws/table_aws_macie2_classification_job.go:177]:
```suggestion
```

File: aws/table_aws_macie2_classification_job.go [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+			return !isLast
+		},
+	)
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAwsMacie2ClassificationJob(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Subhajit97) [inline aws/table_aws_macie2_classification_job.go:188]:
```suggestion
func getMacie2ClassificationJob(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: aws/table_aws_macie2_classification_job.go [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+	plugin.Logger(ctx).Trace("getAwsMacie2ClassificationJob")
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	var id string
+	if h.Item != nil {
+		id = jobId(h.Item)
```
Comment (Subhajit97) [inline aws/table_aws_macie2_classification_job.go:198]: Do we need this?

File: docs/tables/aws_macie2_classification_job.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+
+```sql
+select
+  job_id,
+  detail -> 'AccountId' as account_id,
+  detail -> 'Buckets' as buckets
+from
+  aws_macie2_classification_job,
+  jsonb_array_elements(s3_job_definition -> 'BucketDefinitions') as s3_details,
+  jsonb(s3_details) as detail;
```
Comment (Subhajit97) [inline docs/tables/aws_macie2_classification_job.md:31]: Do we need this?

File: docs/tables/aws_macie2_classification_job.md [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+```
+
+
+### List jobs which are Paused/Cancelled
```
Comment (Subhajit97) [inline docs/tables/aws_macie2_classification_job.md:35]:
```suggestion
### List Paused/Cancelled jobs
```

File: aws/table_aws_macie2_classification_job.go (hunk 11 above)
Comment (bigdatasourav) [reply aws/table_aws_macie2_classification_job.go:186]: There are hydrate columns

File: aws/table_aws_macie2_classification_job.go [hunk 14]
```diff
@@ -0,0 +1,221 @@
...
+			},
+			{
+				Name:        "job_type",
+				Description: "The schedule for running a classification job.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "bucket_definitions",
+				Description: "The namespace of the AWS service that provides the resource, or a custom-resource.",
+				Type:        proto.ColumnType_STRING,
```
Comment (Subhajit97) [inline aws/table_aws_macie2_classification_job.go:64]: Is the column type correct?
Seems like the type should be JSON

File: docs/tables/aws_macie2_classification_job.md [hunk 15]
```diff
@@ -0,0 +1,57 @@
+# Table: aws_macie2_classification_job
+
+An Amazon S3 bucket is a public cloud storage resource available in Amazon Web Services' (AWS) Simple Storage Service (S3), an object storage offering.
```
Comment (Subhajit97) [inline docs/tables/aws_macie2_classification_job.md:3]: Update the description

**Review by Subhajit97 — APPROVED** [review]

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**